core/src/main/scala/kafka/api/FetchRequest.scala

```diff
@@ -196,12 +200,14 @@ case class FetchRequest(versionId: Short = FetchRequest.CurrentVersion,
   }
 
   override  def handleError(e: Throwable, requestChannel: RequestChannel, request: RequestChannel.Request): Unit = {
-    val fetchResponsePartitionData = requestInfo.map { case (topicAndPartition, _) =>
-      (topicAndPartition, FetchResponsePartitionData(Errors.forException(e).code, -1, MessageSet.Empty))
+    val responseData = new util.LinkedHashMap[TopicPartition, JFetchResponse.PartitionData]
```

Comment:
Makes sense. A comment might be useful here to make sure nobody changes it, not realizing the reason for the use of `LinkedHashMap` (unless it's covered by clearly named unit tests, in which case it's probably fine as is).